core/src/main/java/org/elasticsearch/plugins/Plugin.java

```diff
@@ -99,6 +101,12 @@ public Settings additionalSettings() {
     public void onIndexModule(IndexModule indexModule) {}
 
     /**
+     * Called after a search response finishes executing, just prior to being sent back to the client.
+     * Can be used to collect details about the request/response
+     */
+    public void onFinishedSearchResponse(SearchRequest request, SearchResponse response) {}
```

Comment:
Yeah - your right. Whatever path we go down we need to make sure nothing from the plugin package leaks out of bootstrapping.